spec/system/ai_helper/ai_proofreading_spec.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+RSpec.describe "AI Composer Proofreading Features", type: :system, js: true do
+  fab!(:user) { Fabricate(:admin) }
+
+  before do
+    Group.find_by(id: Group::AUTO_GROUPS[:admins]).add(user)
+    assign_fake_provider_to(:ai_helper_model)
+    SiteSetting.ai_helper_enabled = true
+    sign_in(user)
+  end
+
+  let(:composer) { PageObjects::Components::Composer.new }
+
+  it "proofreads selected text using the composer toolbar" do
+    visit "/new-topic"
+    composer.fill_content("hello worldd !")
+
+    # Select a portion of text
+    page.execute_script("document.querySelector('.d-editor-input').setSelectionRange(6, 12);")
```

Comment:
I thought we had a page object method for this but we didn't, added here:

https://github.com/discourse/discourse/pull/28539